canonical/charmhub.io
# Add inks for integrations

## Done
- Ensure `/interfaces/:interface` does not 404 on production
- Add the uuid package and install TypeScript type definitions for uuid
- Include links for interfaces on the integration tab
- Create a custom component to render a mermaid diagram and use it with React Markdown

## How to QA
- Go to https://charmhub-io-1773.demos.haus/kafka/integrations
- Check if you can click on the second part of a title, e.g., `kafka_client` on `kafka-client | kafka_client`
- Check if `See all interfaces` link is hidden in the header
- Check if you can see `Developer Documentation` section rendered including a mermaid diagram

## Issue / Card
Fixes #https://warthogs.atlassian.net/browse/WD-8668

## Screenshot

1. Integrations page
<img width="934" alt="Screenshot 2024-02-29 at 6 43 42 AM" src="https://github.com/canonical/charmhub.io/assets/90341644/f857ced8-b6a6-408e-92ab-e91806a14399">

2. Interface Detail page, e.g., kafka_client
<img width="1265" alt="Screenshot 2024-03-25 at 7 44 17 AM" src="https://github.com/canonical/charmhub.io/assets/90341644/0b9f01a7-ad88-4478-a377-3244a439e7ca">
